### PR TITLE
Fix translation status for outdated mts

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description
The translation status for machine translations used to still show the bot icon even when the translation wasn't up-to-date anymore. This PR fixes this and only shows the bot icon for machine translations that are up-to-date and shows the warning icon for machine translations that are outdated. 

### Proposed changes

- Adds distinction between up-to-date and outdated machine translations and decides for the correct translation status and icon

### Side effects

- I restricted the query for machine translations and this could cause some side effects, although I don't have an exact idea how this might show 

### Resolved issues

Fixes: #2137

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/models/abstract_content_translation.py [hunk 1]
```diff
@@ -447,6 +445,18 @@ def translation_state(self):
             # If the source language does not have a major public version, the translation is considered "outdated",
             # because the content is not in sync with its source translation
             return translation_status.OUTDATED
+        if (
```
Comment (ulliholtgrave) [inline integreat_cms/cms/models/abstract_content_translation.py:448]: I guess you can optimize this a little bit. See suggestion:
```suggestion
        if translation.machine_translated:
            if translation.last_updated >= source_translation.last_updated:
                # If the translation has been made by machine translation and is up to date, show the bot icon
                return translation_status.MACHINE_TRANSLATED
            # If the translation has been made by machine translation but is not up to date, show the outdated icon
            return translation_status.OUTDATED
```

**Review by ulliholtgrave — CHANGES_REQUESTED** [review]
In general, this seems to fix the issue, but I have one small suggestion for an improvement :)

File: integreat_cms/cms/models/abstract_content_translation.py (hunk 1 above)
Comment (timoludwig) [reply integreat_cms/cms/models/abstract_content_translation.py:448]: I agree that it can be optimized, but I would go one step further: We don't even need the nested `if` condition if we just move the "machine translated" check down and use "up to date" as last return value:
```
    def translation_state(self):
        """
        This function returns the current state of a translation in the given language.

        :return: A string describing the state of the translation, one of :data:`~integreat_cms.cms.constants.translation_status.CHOICES`
        :rtype: str
        """
        if not (translation := self.major_version):
            # If the page does not have a major public version, it is considered "missing" (keep in mind that it might
            # have draft versions or public versions that are marked as "minor edit")
            return translation_status.MISSING
        if translation.currently_in_translation:
            return translation_status.IN_TRANSLATION
        if not self.source_language:
            # If the language of this translation is the root of this region's language tree, it is always "up to date"
            return translation_status.UP_TO_DATE
        if not (source_translation := self.major_source_translation):
            # If the source language does not have a major public version, the translation is considered "outdated",
            # because the content is not in sync with its source translation
            return translation_status.OUTDATED
        if source_translation.translation_state == translation_status.OUTDATED:
            # If the source translation is already outdated, this translation is as well
            return translation_status.OUTDATED
        if translation.last_updated < source_translation.last_updated:
            # If the translation was edited before the last major change in the source language, it is outdated
            return translation_status.OUTDATED
        if translation.machine_translated:
            # If the translation has been made by machine translation and is up to date, show the bot icon
            return translation_status.MACHINE_TRANSLATED
        # If the translation was edited after the source translation, we consider it up to date
        return translation_status.UP_TO_DATE
```

File: integreat_cms/release_notes/current/unreleased/2155.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+en: Fix translation status for outdated translations after machine translation
+de: Behebe Fehler beim Übersetzungsstatus für abgelaufener Übersetzung nach machineller Übersetzung
```
Comment (timoludwig) [inline integreat_cms/release_notes/current/unreleased/2155.yml:2]:
```suggestion
de: Behebe Fehler beim Übersetzungsstatus für veraltete maschinelle Übersetzungen
```

**Review by timoludwig — CHANGES_REQUESTED** [review]

File: integreat_cms/cms/models/abstract_content_translation.py [hunk 3]
```diff
@@ -436,25 +436,26 @@ def translation_state(self):
...
             # If the source language does not have a major public version, the translation is considered "outdated",
             # because the content is not in sync with its source translation
-            return translation_status.OUTDATED
-        if source_translation.translation_state == translation_status.OUTDATED:
+            not (source_translation := self.major_source_translation)
             # If the source translation is already outdated, this translation is as well
+            or source_translation.translation_state == translation_status.OUTDATED
+            # If the translation was edited before the last major change in the source language, it is outdated
+            or translation.last_updated < source_translation.last_updated
```
Comment (timoludwig) [inline integreat_cms/cms/models/abstract_content_translation.py:451]:
```suggestion
            or translation.last_updated <= source_translation.last_updated
```
Haha, I thought the case that two translations are created in exactly the same second is a rare enough edge case that we can ignore it - apparently, in the test data there are translations with exactly the same date and they expect the translations to be flagged as outdated in this case...
I'm fine with either way - adapting the tests or make this if-condition compliant with the tests^^

**Review by timoludwig — APPROVED** [review]
:+1:

**Review by ulliholtgrave — APPROVED** [review]
Looks good 👍
